src/mx_bluesky/hyperion/parameters/gridscan.py

```diff
@@ -31,20 +33,30 @@
 class GridCommon(
     DiffractionExperimentWithSample, OptionalGonioAngleStarts, WithOavCentring
 ):
+    use_panda: bool = Field(default=CONST.I03.USE_PANDA_FOR_GRIDSCAN)
+    use_gpu: bool = Field(default=CONST.I03.USE_GPU_FOR_GRIDSCAN_ANALYSIS)
+    use_cpu_and_gpu_zocalo: bool = Field(default=CONST.I03.USE_CPU_AND_GPU_ZOCALO)
     grid_width_um: float = Field(default=CONST.PARAM.GRIDSCAN.WIDTH_UM)
     exposure_time_s: float = Field(default=CONST.PARAM.GRIDSCAN.EXPOSURE_TIME_S)
     use_roi_mode: bool = Field(default=CONST.PARAM.GRIDSCAN.USE_ROI)
     panda_runup_distance_mm: float = Field(
         default=CONST.HARDWARE.PANDA_FGS_RUN_UP_DEFAULT
     )
-    use_panda: bool = Field(default=CONST.I03.USE_PANDA_FOR_GRIDSCAN)
-    use_gpu: bool = Field(default=CONST.I03.USE_GPU_FOR_GRIDSCAN_ANALYSIS)
-    use_cpu_and_gpu_zocalo: bool = Field(default=CONST.I03.USE_CPU_AND_GPU_ZOCALO)
     ispyb_experiment_type: IspybExperimentType = Field(
         default=IspybExperimentType.GRIDSCAN_3D
     )
     selected_aperture: ApertureValue | None = Field(default=ApertureValue.SMALL)
 
+    @model_validator(mode="before")
+    @classmethod
+    def set_default_feature_flags(cls, values) -> Any:
+        cls.features = FeatureFlags()
+        if "use_panda" not in values:
+            values["use_panda"] = cls.features.best_effort().use_panda_for_gridscan
+        if "use_gpu" not in values:
+            values["use_gpu"] = cls.features.best_effort().use_gpu_for_gridscan
```

Comment:
Should: Only fetch the flags from the server once and only define them in one place.

With something like:

```suggestion
    def set_default_feature_flags(cls, values) -> Any:
        cls.features = FeatureFlags.best_effort()
        if "use_panda" in values:
            cls.features.use_panda_for_gridscan = values["use_panda"]
        ...
```

then the new definitions above could be removed?

Reply:
I agree to only get the flags from the server once, but I think the definitions for, eg `use_panda` still need to be in `GridCommon`, just having them in the validator isn't enough for pydantic to know that `GridCommon` has these attributes

Reply:
oh, of course, sorry. but `use_panda` still shouldn't be in `GridCommon`, the flags should be passed in in `features`, which is a `BaseModel`
in the validator for `features` you can then start with FeatureFlags.best_effort() and update anything which is passed in

Reply:
Okay, that works, does that mean we should change the logic of `best_effort`? Right now it checks the URL and uses the constants if it cant find anything. But best efforts should
1. Check if GDA supplied values for each parameter, if yes then use that
2. Check if parameter exists in URL, if yes then use that
3. Use Hyperion constants as a last resort

Is that reasonable?

Reply:
I don't think you need to change the logic in `best_effort` to achieve that priority order, just set the defaults in teh `FeatureFlags` class to the constants. maybe add a new layer to `I03Constants` to hold defaults for feature flags.

Reply:
but if you want to add the ability to override stuff in best_effort with kwargs that sounds fine to me

Reply:
I think we still want GDA parameters to have priority. We use `best_effort` in our plans so these should also give us the GDA parameters when they're supplied

Reply:
I think I've changed my mind actually, I think the config server should take priority once it's actually being used